features/IFC102_Absence-of-deprecated-entities.feature

```diff
@@ -1,74 +1,319 @@
+
+
+  Scenario Outline: Check for deprecated attributes - IFC2X3  
+
+      Given A model with Schema "IFC2X3"
+      Given an <Entity>
+
+      Then <Attribute> = empty
+
+    Examples:
+      | Entity                        | Attribute                     |
+      | IfcFillAreaStyleHatching      | PointOfReferenceHatchLine     |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC4.3
+    # IfcBuildingElementProxy: IfcBuildingElementProxyTypeEnum
+    # IfcFireSuppressionTerminal: IfcFireSuppressionTerminalTypeEnum
+    # IfcCableCarrierFitting: IfcCableCarrierFittingTypeEnum
+    # IfcGeographicElement: IfcGeographicElementTypeEnum
+    # IfcSpace: IfcSpaceTypeEnum
+    # IfcWall: IfcWallTypeEnum
+
+    Given A model with Schema "IFC4.3"
+    Given an <Entity>
+
+    Then PredefinedType is not <Value>
+
+    Examples:
+      | Entity                          |        Value                   |
+      | IfcBuildingElementProxy         | "PROVISIONFORSPACE"            |
+      | IfcBuildingElementProxy         | "PROVISIONFORVOID"             |
+      | IfcFireSuppressionTerminal      | "SPRINKLERDEFLECTOR"           |
+      | IfcFireSuppressionTerminalType  | "SPRINKLERDEFLECTOR"           |
+      | IfcCableCarrierFitting          | "TEE" or "CROSS" or "REDUCER"  |
+      | IfcCableCarrierFittingType      | "TEE" or "CROSS" or "REDUCER"  |
+      | IfcGeographicElement            | "SOIL_BORING_POINT"            |
+      | IfcGeographicElementType        | "SOIL_BORING_POINT"            |
+      | IfcSpace                        | "INTERNAL" or "EXTERNAL"       |
+      | IfcSpaceType                    | "INTERNAL" or "EXTERNAL"       |
+      | IfcStructuralLoadGroup          | "LOAD_COMBINATION"             |
+      | IfcWall                         | "POLYGONAL" or "STANDARD"      |
+      | IfcWallType                     | "POLYGONAL" or "STANDARD"      |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC4
+    # IfcStructuralLoadGroup: IfcLoadGroupTypeEnum
+    # IfcWall: IfcWallTypeEnum
+
+    Given A model with Schema "IFC4"
+    Given an <Entity>
+
+    Then PredefinedType is not <Value>
+
+    Examples:
+      | Entity                   |  Value                                       |
+      | IfcStructuralLoadGroup   | "LOAD_COMBINATION"                           | 
+      | IfcWall                  | "POLYGONAL" or "STANDARD" or "ELEMENTEDWALL" |
+      | IfcWallType              | "POLYGONAL" or "STANDARD" or "ELEMENTEDWALL" |
+
+
+  Scenario Outline: Check for deprecated enumerated values - IFC2X3
+    # IfcOwnerHistory: IfcChangeActionEnum
+
+    Given A model with Schema "IFC2X3"
+    Given an <Entity>
+
+    Then <Attribute> is not <Value>
+
+    Examples:
+      | Entity                   | Attribute            |         Value                                |
+      | IfcOwnerHistory          | ChangeAction         | "MODIFIEDADDED" or "MODIFIEDDELETED"         |
+
+
```

Comment:
This scenario on L253-270 should be removed.  It has the same entities as the following scenario, which is being applied to both IFC4 and IFC4X3.

Reply:
I'd rather be a bit consistent in splitting the schema version (agree, the IFC4.3 then need to be removed from the second scenario). This way we can easily add and remove entities, and it's less prone to errors. 
See also ; https://github.com/buildingSMART/ifc-gherkin-rules/pull/223/commits/58c1c5517c7b81d105c8a2c3a05fdbf5c9b122aa